### PR TITLE
DAOS-1388 Verify snap_list bad parameter behavior (DAOS-1322 DAOS-1392)

### PR DESCRIPTION
**Pull request by dinghwah**
Updated file: src/tests/ftest/container/snapshot.py

Signed-off-by: Ding Ho <ding-hwa.ho@intel.com>

### CODE REVIEW

File: src/tests/ftest/container/snapshot.py [hunk 1]
```diff
@@ -277,6 +296,33 @@ def test_snapshot_negativecases(self):
...
+        #(6)DAOS-1392 Test snapshot with an invalid epoch
+        self.log.info(
+            "==(6)DAOS-1392 destroy snapshot with an invalid handle.")
+        try:
+            snapshot.destroy(None, epoch)
+            self.fail(
+                "##(6)Negative test destroy snapshot with an "
+                "invalid coh handle, expected fail, shown Passing##")
+        except Exception as error:
+            self.log.info(
```
Comment (saurabhtandan) [inline src/tests/ftest/container/snapshot.py]: Can we test for the exact expected ERR to be returned when snapshot destroy fails with an invalid epoch?
Say for example it is expected to fail with ERR -1001, can we check for that and make sure it fails with the same error and not something else.

File: src/tests/ftest/container/snapshot.py [hunk 2]
```diff
@@ -277,6 +296,33 @@ def test_snapshot_negativecases(self):
...
+
+        #(7)DAOS-1388 Verify snap_list bad parameter behavior
+        self.log.info(
+            "==(7)DAOS-1388 Verify snap_list bad parameter behavior.")
+        try:
+            snapshot.list(None, 0)
+            self.fail(
+                "##(7)Negative test snapshot list with an "
+                "invalid coh and epoch, expected fail, shown Passing##")
+        except Exception as error:
```
Comment (saurabhtandan) [inline src/tests/ftest/container/snapshot.py]: Same as above this and all other such cases in this test where test is expected to fail.

Comment (dinghwah) [reply src/tests/ftest/container/snapshot.py]: Thanks for your comment.
Yes.
Snapshot negative test with invalid epoch or snapshot list handle will generate Error of RC: -1002.
Testlog:
==(6)DAOS-1392 destroy snapshot with an invalid handle.
==>Negative test, destroy snapshot with an invalid handle.
   Expected Error: Failed to destroy the snapshot. RC: -1002
==(7)DAOS-1388 Verify snap_list bad parameter behavior.
==>Negative test, snapshot list with an invalid coh.
   Expected Error: Snapshot create returned non-zero. RC: -1002

File: src/tests/ftest/container/snapshot.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/container/snapshot.py]: Yes.
invalid snapshot epoch will generate Error of RC: -1002.

File: src/tests/ftest/container/snapshot.py (hunk 2 above)
Comment (dinghwah) [reply src/tests/ftest/container/snapshot.py]: Please see the new script with exact error verification.

File: src/tests/ftest/container/snapshot.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/container/snapshot.py]: Please see the newly committed script for the update.

**Pull request merged**